src/lib/interactionManager/authenticationFlow.ts

```diff
@@ -18,8 +18,12 @@ export class AuthenticationFlow extends Flow<Authentication> {
     token: Authentication,
     interactionType: string,
   ) {
+    // FIXME what's with this multilayer type checking
     switch (interactionType) {
       case InteractionType.Authentication:
```

Comment:
Can we use a guard here, i.e. case `isAuthenticationRequest`, to make TS happy (otherwise we have to cast the token to to expected type).

Reply:
Not sure if TS picks this up, i.e. a guard in the switch condition.

Reply:
there's a bit of a long story here, and I think solving it requires another refactor that I would like to leave for another time 😄

Reply:
It's interacting with the issue of `interactionType` basically being missing from the payload... we could have a much easier time with pleasing typescript if only the `interactionType` was actually inside the token payload